### DISCUSSION TITLE
Love the way you put your idea behind assembler css

### DISCUSSION THREAD

**Discussion by leamandeep** [asmcss/assembler#31] [Show and tell]
Hi Assembler Team! Loved the idea behind Assembler CSS. Actually , I am working on my React UI library and I don't want to use CSS in JS library. So, I am looking towards utility frameworks  like Tailwind CSS. The problem with Tailwind CSS we can't use sass like feature. So , i thought of using something like sass and I found Assembler CSS. I will soon going to learn and implement into my React UI Library. 

 Few questions wandering in my mind  👇
> Will it works for JSX?

> Final bundle size of final css code 

> How will I use finale css code?

> Any Concept of grouping

**Comment by msarca** [comment]
>> Will it works for JSX?

There is no reason it shouldn't, but you must be careful when using frameworks like Vue or React not to combine `x-style` with reactive `style` attributes. Also, you should checkout [this setting](https://asmcss.com/docs/1.x/configuration.html#data-selector-attribute) if you plan to combine `x-style` with reactive `class` attributes.


> 
> > Final bundle size of final css code
> 
> > How will I use finale css code?
> 
> > Any Concept of grouping

No files are generated, so there is no concept of bundles in Assembler CSS. You only need to include a single JavaScript file, as stated in the [documentation](https://asmcss.com/docs/1.x/#installation). Assembler CSS is a truly just-in-time framework and generates CSS rules on the fly. The number of rule varies from page to page: for our [landing page](https://asmcss.com) only 73 CSS rules are generated; but for a more complex page like [Expressive Suite's](https://www.expressivesuite.com) landing page, 137 rules are generated.

**Comment by saidwho12** [comment]
Hi, I really like this project idea and would love to contribute to it. Seems like what I always wanted when writing webpages. Inline shortened CSS 
with helper macros/functions is definitely the better approach to styling. Looked at Tailwind CSS but sadly it is not really well fit for my project, since I am using Go for backend and Tailwind CSS is built for NodeJS (yuck!). 

I also tried to hand-write widget classes and use use SASS to help shorten the code, still ends up being bigger than what is reasonable for maintaining fast load times on my pages. The advantages of JIT is that you can edit your layout structure and styling without switching context, and there is no CSS file at all that the client has to send, everything can be embedded, using this with HTML templating is very efficient and comfortable.

As someone coming from C and lower level languages, and frankly don't like to mess with too many web technologies, I love this. I like the performance of this bet it could be really good if it used wasm as well.
